git_repo/repo.py

```diff
@@ -434,9 +434,15 @@ def request_edition(repository, from_branch, onto_target):
                 self.message,
                 self._auto_slug,
                 request_edition)
-        log.info('Successfully created request of `{local}` onto `{project}:{remote}`, with id `{ref}`!'.format(**new_request))
+        if 'ref' in new_request:
+            log.info('Successfully created request of `{local}` onto `{project}:{remote}`, with id `{ref}`!'.format(**new_request))
+        else:
+            log.info('Successfully created request of `{local}` onto `{project}:{remote}`!'.format(**new_request))
         if 'url' in new_request:
-            log.info('available at: {url}'.format(**new_request))
+            if isinstance(new_request['url'], list):
```

Comment:
ah! this is not very elegant… What value does adding checking the type of the URL here? I can see a few ways to get rid of that issue, either:

* make the `request_create` method return a string `"\n{}".format('\n'.join(…))`, or
* make the `request_create` method always return a list, and modify the other services to match that, or
* make the `do_request_create` method delegate the way it displays stuff the same way we do with `list` methods (make that method a generator, where you first yield a format, then a header, then the values).

(I sorted the solution by order of laziness, the first one being the less work ☺)

Though, I believe the third option would be the most elegant, so that way you could control the exact message you want to output to the user to make git-repo more adapted to the gerrit way of things!